src/cloudai/workloads/common/llm_serving.py

```
@@ -427,13 +427,41 @@ def generate_wait_for_health_function(self) -> str:
+    i=0
+    while kill -0 "${pid_var}" 2>/dev/null; do
+        [ "$i" -ge {timeout} ] && echo "PID did not exit in time" && return 1
+        sleep 1
+        i=$((i+1))
+    done
+}}
+trap cleanup EXIT"""
+
         pid_values = " ".join(f"{pid_var}=${pid_var}" for pid_var in pid_vars)
-        kill_lines = "\n".join(f'    [ -n "${pid_var}" ] && kill -9 ${pid_var} 2>/dev/null' for pid_var in pid_vars)
+        pid_array = " ".join(map(lambda p: f'"${p}"', pid_vars))
```

Ruff (0.15.10)

[warning] Unnecessary `map()` usage (rewrite using a generator expression)

Replace `map()` with a generator expression

(C417)